### PR TITLE
Work for SLES 12 compatibility

### PR DESCRIPTION
**Pull request by HelenCampbell**

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -156,6 +156,8 @@
               $service_name  = 'ntpd'
               $keys_file     = '/etc/ntp.keys'
               $package_name  = $default_package_name
+              #SLES12 uses sytemd as its service provider, hence why we hardcode it here.
```
Comment (bmjen) [inline manifests/params.pp]: `systemd`

File: manifests/params.pp [hunk 2]
```diff
@@ -156,6 +156,8 @@
               $service_name  = 'ntpd'
               $keys_file     = '/etc/ntp.keys'
               $package_name  = $default_package_name
+              #SLES12 uses sytemd as its service provider, hence why we hardcode it here. 
+              $service_provider = 'systemd'
```
Comment (bmjen) [inline manifests/params.pp]: Personal preference: I'd prefer the `$service_provider` to be set more explicitly... so it should be set to `undef` on all of the other cases. Similar to how `$maxpoll` is handled. It's pretty obvious to the majority of people reading the code that `$service_provider` should be `undef` in all platforms != SLES12, but it doesn't hurt to be explicit IMO.

**Comment by bmjen** [comment]
Sorry @HelenCampbell. Another nit. Can you add a spec test to validate that the `$service_provider` is set correctly for SLES12 by checking to see if the `ntp_config` file is deleted and also another test that the param is set correctly when someone explicitly passes in a `$service_provider`?

**Comment by HelenCampbell** [comment]
@DavidS @bmjen Test updates are in

**Pull request merged**